### PR TITLE
Update which signals are handled by cleanup.

### PR DESCRIPTION
**Pull request by stephen-soltesz**
This change removes the unconditional loop that registers the 'cleanup'
signal handler for *all* signals and registers the handler for
specific signals that perform some action in cleanup.

This is to take advantage of the default implementations for SIGINT,
SIGTERM, and SIGTSTP and to avoid unnecessary handling of all other
signals.

### CODE REVIEW

File: src/web100srv.c [hunk 1]
```diff
@@ -307,9 +307,6 @@ void cleanup(int signo) {
 
   sigsafe_debug_log(1, signo, "Received a signal");
   switch (signo) {
-    default:
```
Comment (pboothe) [inline src/web100srv.c]: We should keep the `default:` case. That way if rogue signals happen we will get log messages about them.

File: src/web100srv.c [hunk 2]
```diff
@@ -2007,7 +1995,7 @@ SSL_CTX *setup_SSL(const char *certificate_file, const char *private_key_file) {
  */
 int main(int argc, char **argv) {
   int c, i;
-  struct sigaction new;
+  struct sigaction mysignal_action;
```
Comment (pboothe) [inline src/web100srv.c]: NO

File: src/web100srv.c [hunk 3]
```diff
@@ -2421,14 +2409,19 @@ int main(int argc, char **argv) {
...
+  //   SIGTSTP (ctrl-z), SIGINT (ctrl-c), or SIGTERM (default kill signal).
+  // Only register handlers for signals acted on in `cleanup`.
+  memset(&mysignal_action, 0, sizeof(mysignal_action));
+  mysignal_action.sa_handler = cleanup;
+
+  sigaction(SIGALRM, &mysignal_action, NULL);
+  sigaction(SIGCHLD, &mysignal_action, NULL);
+  sigaction(SIGHUP, &mysignal_action, NULL);
+  sigaction(SIGPIPE, &mysignal_action, NULL);
+  sigaction(SIGSEGV, &mysignal_action, NULL);
```
Comment (pboothe) [inline src/web100srv.c]: Put a comment on this one please, about how we shouldn't need this.

File: src/web100srv.c (hunk 2 above)
Comment (stephen-soltesz) [inline src/web100srv.c]: Muahahaha!!

File: src/web100srv.c (hunk 3 above)
Comment (stephen-soltesz) [inline src/web100srv.c]: Done.

File: src/web100srv.c (hunk 1 above)
Comment (stephen-soltesz) [inline src/web100srv.c]: Done.

**Comment by stephen-soltesz** [comment]
PTAL?

**Comment by pboothe** [comment]
LGTM

**Pull request merged**